common/hooks/useSpotify.tsx

```diff
@@ -42,12 +42,23 @@ interface SpotifyProviderProps {
 }
 
 export const SpotifyProvider: React.FunctionComponent<SpotifyProviderProps> = (props) => {
-    const { accessToken, user } = useAuth()
-    const { notifyError, notifySuccess } = useNotification()
-    const spotifyHelper = new SpotifyHelper(accessToken)
+    const { user, getNewTokensFromRefreshToken } = useAuth()
+    let { accessToken, refreshToken } = useAuth()
```

Comment:
i actually had to make those tokens let because i found that i was forced to update them to make everything work. if the access token expires, i get a new one and return in, update the access token for the context and make a new spotify helper variable by passing in the new access token; it was the only way i could get it to work. i thought that if i set the new access token in the auth context then that would update it everywhere including the spotify context, but it didn't. so, long story short, i return that new access token and then use that to continue the flow of it all. i do suppose that the refresh doesn't need to be let so ill update that.

Reply:
look at comment below about this, shouldn't be updating context state variables without a setter